### DISCUSSION TITLE
watch, useFormContext and setValue usage

### DISCUSSION THREAD

**Discussion by bombillazo** [react-hook-form/react-hook-form#8644] [Q&A] [answered]
Hello! I am trying to use `useFormContext()` to use `watch` and `setValues` to set and react to a form value from different components.  When I use setValues to set the value of one of the form fields, however, the other component "listening" to that value using watch does not trigger a rerender or detect the value it is watching has changed. Am I missing something in my logic?

**Comment by bluebill1049** [comment, marked as answer]
share a simple codesandbox?

**Reply by bombillazo** [reply]
Nevermind, it is actually working, it's one of my components that is not reacting to the value changes :S

But thanks for the swift response!!!

**Reply by bluebill1049** [reply]
Awesome.
